### PR TITLE
Enable TIAF Native Tests Phase 2

### PR DESCRIPTION
**Pull request by jonawals**
## What does this PR do?

As per the title. Additionally, as Phase 2 is being enabled before Phase 1, this PR aso enables the native TIAF runtime.

Note: Some of the longpole tests in Phase 1 are taking an extraordinary length of time due to the OpenCppCoverage instrumentation (e.g. AZ::AzToolsFramework.Tests takes approximately 20 mins to complete with OpenCppCoverage instrumentation, as opposed to approximately 4 minutes without). With the optimizations introduced by the native test sharding optimiation, AZ::AzToolsFramework.Tests takes approximately 2 mins with OpenCppCoverage, making these optimizations critical for rolling out Phase 1. As such, Phases 2 and 3 will be rolled out first and, once the native test sharding optimizations are complete, Phase 1 will be rolled out.

## How was this PR tested?

This PR will be tested in AR.

### CODE REVIEW

File: scripts/build/Platform/Windows/build_config.json
Comment (moraaar) [inline scripts/build/Platform/Windows/build_config.json:210]: question: Is TIAF windows only? What about Linux?

**Review by moraaar — APPROVED** [review]

File: scripts/build/Platform/Windows/build_config.json
Comment (jonawals) [reply scripts/build/Platform/Windows/build_config.json:210]: TIAF is Windows only for now. It would be relatively straightforward to support Linux, but it would require implementing the necessary platform and toolchain specific support as there is no common test coverage tool on both platforms or compiler support for coverage between all supported compilers on both platforms, hence the decision to focus on Windows to demonstrate the capabilities before considering Linux support.

**Review by hultonha — APPROVED** [review]
LGTM! Thanks for the detailed description too 👍 (might be worth waiting until Monday to merge just in case there's any (albeit unlikely) fallout)

**Pull request merged**
